lib/train/transports/local.rb

```diff
@@ -241,25 +248,25 @@ def acquire_pipe
           pipe
         end
 
+        # 5. Set strict ACLs on named pipes (PowerShell)
         def start_pipe_server(pipe_name)
           require "win32/process"
-
           script = <<-EOF
             $ErrorActionPreference = 'Stop'
-
-            $pipeServer = New-Object System.IO.Pipes.NamedPipeServerStream('#{pipe_name}')
+            $pipeSecurity = New-Object System.IO.Pipes.PipeSecurity
+            $rule = New-Object System.IO.Pipes.PipeAccessRule("#{ENV["USERNAME"]}", "FullControl", "Allow")
```

Comment:
Will the `USERNAME` be set by the user themselves, or is there something I am missing here? 

In case we are asking for USERNAME from the user:
- Not sure if we should ask user for this.
- Also for this logic to work, ENV value of username should match with `owner` and `current_user` values right?
And `current_user` and `owner` might return values in the format `"DOMAIN\\Username"`. Just making sure that we communicate this to the user to set USERNAME is in this format.